### PR TITLE
Expand CMake documentation a bit

### PR DESCRIPTION
**Pull request by madebr**
Expand CMake documentation a bit

Does NOT solve #5237, because it does not address Android.

I'm planning to greatly simplify bootstrapping new Android SDL applications, which would nullify changes made now.

Pleaes proofread this

### CODE REVIEW

**Comment by ericoporto** [comment]
@madebr , about the issue and CMake docs, Android is the only platform that I mentioned needs **no** changes because it already is written from the user project perspective.

**Comment by slouken** [comment]
The documentation changes look good to me. Feel free to merge when you're ready.

**Comment by icculus** [comment]
I resolved the conflict, but some of the builds are failing with:

```
CMake Error at /home/runner/work/SDL/SDL/autotools_prefix/lib/cmake/SDL2/sdl2-config.cmake:13 (message):
  File or directory /home/runner/work/SDL/SDL/autotools_prefix/lib/bin
-- Configuring incomplete, errors occurred!
  referenced by variable SDL2_BINDIR does not exist !
```

I don't know it that's this patch or something else yet.

File: sdl2-config.cmake.in [hunk 1]
```diff
@@ -14,18 +14,8 @@ macro(set_and_check _var _file)
...
-macro(check_required_components _NAME)
-    foreach(comp ${${_NAME}_FIND_COMPONENTS})
-        if(NOT ${_NAME}_${comp}_FOUND)
-            if(${_NAME}_FIND_REQUIRED_${comp})
-                set(${_NAME}_FOUND FALSE)
-            endif()
-        endif()
-    endforeach()
-endmacro()
+get_filename_component(prefix "${CMAKE_CURRENT_LIST_DIR}@cmake_prefix_relpath@" ABSOLUTE)
```
Comment (madebr) [inline sdl2-config.cmake.in:17]:
```suggestion
get_filename_component(prefix "${CMAKE_CURRENT_LIST_DIR}/@cmake_prefix_relpath@" ABSOLUTE)
```

Comment (icculus) [reply sdl2-config.cmake.in]: Whoops, sorry about that!

**Pull request merged**